specs/protocol/holocene/derivation.md

```diff
@@ -1,13 +1,332 @@
+the current sequencing epoch,
+- by an empty list otherwise.
+
+As before, a failure to then process the deposit-only attributes is a critical error.
+
+If an invalid payload is replaced by a deposit-only payload, for consistency reasons, the remaining
+span batch, if applicable, and channel it originated from are dropped as well.
+
+## Activation
+
+The new batch rules activate when the _L1 inclusion block timestamp_ is greater or equal to the
+Holocene activation timestamp. Note that this is in contrast to how span batches activated in
+[Delta](../delta/overview.md), namely via the span batch L1 origin timestamp.
+
+When the L1 traversal stage of the derivation pipeline moves its origin to the L1 block whose
+timestamp matches the Holocene activation timestamp, the derivation pipeline's state is mostly reset
+by **discarding**
+
+- all frames in the frame queue,
+- channels in the channel bank, and
+- all batches in the batch queue.
+
+The three stages are then replaces by the new Holocene frame queue, channel bank and batch queue
+(and, depending on the implementation, the optional span batch stage is added).
+
+Note that batcher implementations must be aware of this activation behavior, so any frames of a
+partially submitted channel that were included pre-Holocene must be sent again. This is a very
+unlikely scenario since production batchers are usually configured to submit a channel in a single
+transaction.
+
+## Sync Start
+
+Thanks to the new strict frame and batch ordering rules, the sync start algorithm can be simplified in the
+average case. The rules guarantee that
+
+- an incoming first frame for a new channel leads to discarding previous incomplete frames for a
+non-closed previous channel in the frame queue and channel bank, and
+- when the derivation pipeline L1 origin progresses, the batch queue is empty.
+
+So the sync start algorithm can optimistically select the last L2 unsafe, safe and finalized heads
+from the engine and if the L2 safe head's L1 origin is _plausible_ (see the
+[original sync start description](../derivation.md#finding-the-sync-starting-point) for details),
+start deriving from this L1 origin.
+
+- If the first frame we find is a _first frame_ for a channel that includes the safe head (TODO: or
+even just the following L2 block with the current safe head as parent?), we can
+safely continue derivation from this channel because no previous derivation pipeline state could
+have influenced the L2 safe head.
+- If the first frame we find is a non-first frame, then we need to go back a full channel
+timeout window to see if we find the start of that channel.
+  - If we find the starting frame, we can continue derivation from it.
+  - If we don't find the starting frame, we need to go back a full sequencing window. (TODO: verify)
+
+_TODO: above needs to be verified, we don't introspect batcher transactions and frame metadata
+during sync start yet._
```

Comment:
I touched on the TODOs in this section in discussions with @protolambda and @clabby, curious what you make of it.

Reply:
I'm actually not sure this needs to be in the spec to be honest.  It's perfectly valid to always start a full ChannelTimeout earlier and the optimisation to optimistically start from the L1 origin and only go back if a mid-channel frame is found is an implementation detail.  Though our specs already blur spec and implementation a fair bit...

Reply:
That's true, it certainly doesn't need to be in the spec. It's sufficient to only contain a _working_ version of the sync start, that doesn't necessarily need to be optimal. However, I still think it can be beneficial to include some high level implementation pointers in the spec. We could just move it into the "Security and Implementation Considerations" section.

Reply:
moved to implementation section